ZEngine/ZEngine/Core/Maths/Matrix.h

```diff
@@ -0,0 +1,485 @@
+#include <Helpers/MemoryOperations.h>
+#include <cstddef>
+#include "Vec.h"
+
+using namespace ZEngine::Helpers;
+namespace ZEngine::Core::Maths
+{
+    template <typename T, size_t R, size_t C, typename = std::enable_if_t<std::is_arithmetic_v<T> && (R >= 1) && (C >= 1)>>
+    struct Matrix
+    {
+        T  m_data[R * C];
+
+        T& operator()(size_t row, size_t col)
+        {
+            ZENGINE_VALIDATE_ASSERT(row < R && col < C, "Index out of range");
+            return m_data[row * C + col];
+        }
+
+        const T& operator()(size_t row, size_t col) const
+        {
+            ZENGINE_VALIDATE_ASSERT(row < R && col < C, "Index out of range");
+            return m_data[row * C + col];
```

Comment:
We want colum-major storage
```suggestion
            return m_data[col * R + row];
```